docker/setup/main.py

```diff
@@ -72,8 +72,12 @@ async def proxy_receive_mediator_invite(mediator: Client, invite: str):
 
 async def get_proxy_invite() -> dict:
     async with AsyncClient() as client:
-        r = await client.get(f"{PROXY}/retrieve_agent_invitation")
-        url = r.json()["invitation_url"]
+        url = None
+        while url is None:
+            r = await client.get(f"{PROXY}/retrieve_agent_invitation")
+            url = r.json()["invitation_url"]
+            if not url:
+                await asyncio.sleep(1)
```

Comment:
Wasn't expecting to see changes to the setup script; perhaps this belongs in another PR?